Submitted by: Heath Caldwell <hncaldwell@gentoo.org>
Date: 2008-10-21
Initial Package Version: 0.10
Upstream Status: Accepted to be included in next release
Description: Allows the leftmost queue to have an escalation number if it is the only queue mentioned.

diff -ur tenshi-0.10/tenshi tenshi-0.10.new/tenshi
--- tenshi-0.10/tenshi	2008-10-06 16:55:37.000000000 -0700
+++ tenshi-0.10.new/tenshi	2008-10-06 17:18:07.000000000 -0700
@@ -522,7 +522,7 @@
                 }
             }
 
-            if ($queue[0] =~ /:/) {
+            if (@queue > 1 and $queue[0] =~ /:/) {
                 die RED "[ERROR] Left most queue in a multiple queue declaration can not have an escalation number\n";
             }
 
diff -ur tenshi-0.10/tenshi.8 tenshi-0.10.new/tenshi.8
--- tenshi-0.10/tenshi.8	2008-03-13 04:31:12.000000000 -0700
+++ tenshi-0.10.new/tenshi.8	2008-10-06 17:09:36.000000000 -0700
@@ -337,16 +337,16 @@
 regexp. The queue will receive the message that matched the regexp at the time
 of escalation, with a count equal to the escalation number. The count of
 messages matching the regexp will be reset when the left most queue mentioned
-in the queue list is mailed.The left most queue cannot have an escalation
-number. When the number of messages that match the regexp reaches the greatest
-escalation number mentioned, escalation will begin again into the escalation
-queues, modulus the greatest escalation number. For example, using the queues
-`a,b:10,c:50', when 10 messages match the regexp, a message will go into b,
-when 50 match, one will go into c. At 60, another will go into b, and at 100,
-another into c, 110 to b, 150 to c, and so on. Escalation numbers must be
-positive integers greater than zero and must be listed in increasing order from
-left to right. All queues without escalation numbers must be listed more left
-than the queues with escalation numbers.
+in the queue list is mailed. The left most queue cannot have an escalation
+number unless it is the only queue listed. When the number of messages that
+match the regexp reaches the greatest escalation number mentioned, escalation
+will begin again into the escalation queues, modulus the greatest escalation
+number. For example, using the queues `a,b:10,c:50', when 10 messages match the
+regexp, a message will go into b, when 50 match, one will go into c. At 60,
+another will go into b, and at 100, another into c, 110 to b, 150 to c, and so
+on. Escalation numbers must be positive integers greater than zero and must be
+listed in increasing order from left to right. All queues without escalation
+numbers must be listed more left than the queues with escalation numbers.
 
 .br
 The standard grouping operators
